test/test_common/utility.cc

```diff
@@ -238,6 +238,26 @@ AssertionResult TestUtility::waitForGaugeEq(Stats::Store& store, const std::stri
   return AssertionSuccess();
 }
 
+AssertionResult TestUtility::waitForGaugeDestroyed(Stats::Store& store, const std::string& name,
+                                                   Event::TestTimeSystem& time_system,
+                                                   std::chrono::milliseconds timeout) {
+  Event::TestTimeSystem::RealTimeBound bound(timeout);
+  while (findGauge(store, name) != nullptr) {
+    time_system.advanceTimeWait(std::chrono::milliseconds(10));
+    if (timeout != std::chrono::milliseconds::zero() && !bound.withinBound()) {
+      std::string current_value;
+      if (findGauge(store, name)) {
+        current_value = absl::StrCat(findGauge(store, name)->value());
+      } else {
```

Comment:
this block of code is not reachable, right? None of the call-sites specify a non-zero timeout?

Reply:
I think it's technical possible since it's multi-thread.

However, I think the issue is that too many findGauge() is invoked. I updated to call once in each loop